src/java/org/apache/cassandra/locator/ReplicaList.java

```diff
@@ -332,4 +330,41 @@ public static ReplicaList toDummyList(Collection<Range<Token>> ranges)
                                .collect(Collectors.toList()));
 
     }
+
+    public void prioritizeForRead()
```

Comment:
So the only thing here when prioritizing is that we need to take into account the magnitude of badness of transient replicas. We shouldn't avoid full replicas if the transient replicas are sufficiently bad. This is of course very hard to quantify, but maybe we can do something like use P50 or P99 or something to select from two sublists, the first made up of "good nodes" and the next made up of bad nodes. We only select bad nodes if we have exhausted "good" nodes.

Or maybe there is some relationship we look for across the scores to sort on where we are biased towards transient but if it's too far from the mean we will sort it elsewhere.

Given time constraints maybe we should just make a note for a followup ticket since this only hurts people using transient replication.

Reply:
Are you sure we want this heuristic?  If the transient nodes are 'bad' then one or more replicas are probably also bad, and a data request from such a replica is going to still be costlier than a data request from the transient replica?

I would assume any dynamic snitch based heuristics would be incorporated to a decision on whether to consult a transient replica or a regular replica, and perhaps the dynamic snitch could maintain separate statistics for the portions that are transient vs full, but this is probably sufficient and more logically straightforward?